### ISSUE TITLE
Failure on `apm update`

### ISSUE THREAD

**Issue by yajo** [FZUG/repo#152]
See the whole story in https://github.com/atom/settings-view/issues/766#issuecomment-230719711 and onwards

**Comment by 1dot75cm** [comment]
Hi @Yajo 
Make sure to install the following package:
- [nodejs-atom-package-manager-1.10.0-6.git87b4bcb.fc23.x86_64.rpm](https://copr-be.cloud.fedoraproject.org/results/mosquito/atom/fedora-23-x86_64/00347481-nodejs-atom-package-manager/nodejs-atom-package-manager-1.10.0-6.git87b4bcb.fc23.x86_64.rpm)
- [electron-legacy-0.37.8-2.prebuilt.fc23.x86_64.rpm](https://copr-be.cloud.fedoraproject.org/results/mosquito/atom/fedora-23-x86_64/00363700-electron-legacy/electron-legacy-0.37.8-2.prebuilt.fc23.x86_64.rpm)
- [atom-1.8.0-2.gitf89b273.fc23.x86_64.rpm](https://copr-be.cloud.fedoraproject.org/results/mosquito/atom/fedora-23-x86_64/00346724-atom/atom-1.8.0-2.gitf89b273.fc23.x86_64.rpm)

The `/usr/lib/node_modules/atom-package-manager/bin/apm` need to read version of electron (`/bin/electron -v`). Otherwise, return the following error:

```
gyp ERR! stack Error: Invalid version number: undefined
```

**Comment by yajo** [comment]
Yes, it seems fixed now. Thanks!

**Comment by bugabinga** [comment]
@1dot75cm whats the difference between electron and electron-legacy? do I need legacy when electron is installed?

**Comment by 1dot75cm** [comment]
The `electron` is a latest version. Some other software need to use(e.g., `vscode`). The `electron-legacy` is a stable version(`0.37.8`).  Only `atom` use it. @flasheater 

Why not use latest version of Electron?  Please see #168.

**Comment by xenithorb** [comment]
Also, it turns out `electron` needs a display in order to get the version ( :-/ ) 

I ran into this doing updates over SSH and the command fails in the manner described above _exactly_ because `electron -v` outputs nothing at all when used over SSH because `DISPLAY=` is not exported. 

```bash
DISPLAY=:0 electron -v
```
Works and shows you the proper electron version, which also means that:

```bash
DISPLAY=:0 apm <command> 
```

allows you to resume what you're doing without error.

**Comment by floriangosse** [comment]
I have the same problem on F23. I have found that the problem is at [/rpms/atom/apm.js#L10](/FZUG/repo/blob/master/rpms/atom/apm.js#L10) and in [/rpms/electron/electron-prebuild.spec](/FZUG/repo/blob/master/rpms/electron/electron-prebuild.spec).

Problem: `atom` requires the latest version of electron which will be installed at `/usr/bin/electron-1.3.13` but `apm.js` expects it at `/bin/electron`.

Solution: I think if `electron` create a sym link to `/usr/bin/electron` and `apm` use this binary the problem should be fixed.

**Comment by xenithorb** [comment]
It does create a symlink. Check `rpm -q --scripts electron`. 

So this problem has something to do with the way the package handles alternatives. For whatever reason, perhaps due to switching between this atom, and the official atom, I also had this problem due to the alternatives links not pointing to the most current version of electron. (They were still pointed at `1.3.9` iirc. I worked around it by removing all of electron `rpm -e --no-deps electron` and then `find / -name 'electron'` and removed the files that shouldn't have been there (most of which, again iirc, were bad symlinks) 

The bottom line is that the electron package needs to be a little more robust about handling the /usr/bin/electron and /bin/electron symlinks, but I don't personally know ~how~ [the right way] to fix that right now.

**Comment by xenithorb** [comment]
@1dot75cm ^ This likely affects all Fedora users given the pattern.

**Comment by 1dot75cm** [comment]
I think it's a good idea to change the priority of `/usr/bin/electron-xxx`. It always links to the current version of electron (if you use `alternatives --auto electron` command). @xenithorb

https://github.com/FZUG/repo/blob/master/rpms/electron/electron-prebuild.spec#L66

**Comment by alick** [comment]
Closing as fixed.
